docs/aws/audit/cloudtrailmonitoring/rules/cloudtrail_s3_bucket_logging.mdx

```diff
@@ -23,6 +23,260 @@ HIPAA, GDPR, SOC2, CISAWS, CBP, AWSWAF
+
+2. List all CloudTrail trails: Use the following AWS CLI command to list all the CloudTrail trails in your AWS account:
+
+   ```
+   aws cloudtrail describe-trails
+   ```
+   This command will return a JSON output with details of all the trails.
+
+3. Check if Server Access Logging is enabled: For each trail, check the 'IsLogging' attribute. If the value of 'IsLogging' is 'true', then Server Access Logging is enabled for that trail. If the value is 'false', then Server Access Logging is not enabled.
+
+   You can use the following command to check the 'IsLogging' attribute for a specific trail:
+
+   ```
+   aws cloudtrail get-trail-status --name <trail_name>
+   ```
+   Replace `<trail_name>` with the name of your trail.
+
+4. Automate the process with a Python script: If you have multiple trails and want to automate the process, you can write a Python script using the boto3 library to list all trails and check if Server Access Logging is enabled for each trail. Here is a simple example:
+
+   ```
+   import boto3
+
+   client = boto3.client('cloudtrail')
+
+   response = client.describe_trails()
+
+   for trail in response['trailList']:
+       status = client.get_trail_status(Name=trail['Name'])
+       print(f"Trail: {trail['Name']}, Logging: {status['IsLogging']}")
+   ```
+   This script will print the name of each trail and whether Server Access Logging is enabled or not.
+</Accordion>
+
+<Accordion title='Using Python'>
+1. Install the necessary Python libraries: Before you can start writing the script, you need to install the necessary Python libraries. The AWS SDK for Python (Boto3) allows Python developers to write software that makes use of services like Amazon S3, Amazon EC2, etc. You can install it using pip:
+
+```python
+pip install boto3
+```
+
+2. Import the necessary libraries and create a session: After installing the necessary libraries, you need to import them into your Python script. You also need to create a session using your AWS credentials.
+
+```python
+import boto3
+
+session = boto3.Session(
+    aws_access_key_id='YOUR_ACCESS_KEY',
+    aws_secret_access_key='YOUR_SECRET_KEY',
+    region_name='us-west-2'
+)
+```
+
+3. Create a CloudTrail client: After creating a session, you need to create a CloudTrail client. This client will allow you to interact with the CloudTrail service.
+
+```python
+cloudtrail = session.client('cloudtrail')
+```
+
+4. Check if Server Access Logging is enabled: Now that you have a CloudTrail client, you can use it to check if Server Access Logging is enabled. You can do this by calling the describe_trails method and checking the S3BucketName attribute.
+
+```python
+response = cloudtrail.describe_trails()
+
+for trail in response['trailList']:
+    if 'S3BucketName' not in trail:
+        print(f"CloudTrail {trail['Name']} does not have Server Access Logging enabled.")
+    else:
+        print(f"CloudTrail {trail['Name']} has Server Access Logging enabled.")
+```
+
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>